src/ResourceManagement/Batch/Microsoft.Azure.Management.Batch.Fluent/ApplicationPackageImpl.cs

```diff
@@ -51,7 +51,7 @@ public string Id
 
         public override string Name()
         {
-            return ResourceUtils.NameFromResourceId(Inner.Id);
+            return Inner.Version;
         }
 
```

Comment:
@anudeepsharma @anuchandy but in `ExternalChildResource` it returns innter.name and not inner.version, right?